### PR TITLE
Add docs for CA2019 and CA2259

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #29264 

[CA2019 preview](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2019?branch=pr-en-us-32207)
[CA2259 preview](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2259?branch=pr-en-us-32207)

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca2019.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+```csharp
+class C
+{
+    [ThreadStatic]
+    private static Object obj;
+
+    static void S1()
+    {
+        obj = new Object();
```
Comment (stephentoub) [inline docs/fundamentals/code-analysis/quality-rules/ca2019.md:63]:
```suggestion
        obj ??= new Object();
```

File: docs/fundamentals/code-analysis/quality-rules/ca2019.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+```
+
+```vb
+Class C
+    <ThreadStatic>
+    Private Shared obj
+
+    Shared Sub S1()
+        obj = New Object()
```
Comment (stephentoub) [inline docs/fundamentals/code-analysis/quality-rules/ca2019.md:74]:
```suggestion
        If obj Is Nothing Then
            obj = New Object()
        End If
```

File: docs/fundamentals/code-analysis/quality-rules/ca2259.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+```vb
+Class C
+    <ThreadStatic>
+    Public number As Integer = 404
+End Class
+```
+
+## When to suppress warnings
+
+It's safe to suppress a warning from this rule.
```
Comment (stephentoub) [inline docs/fundamentals/code-analysis/quality-rules/ca2259.md:55]: You can suppress it, but the [ThreadStatic] is a nop and isn't doing what you intend it to do.

**Review by stephentoub — APPROVED** [review]
Thanks!

File: docs/fundamentals/code-analysis/quality-rules/ca2259.md (hunk 3 above)
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2259.md:55]:
```suggestion
It's safe to suppress a warning from this rule. However, the `[ThreadStatic]` attribute is a nop and isn't doing what you intended it to do.
```

**Pull request merged**